iati/core/tests/test_rulesets.py

```diff
@@ -220,32 +240,44 @@ def test_rule_paths(self, basic_rule):
                 assert path.startswith(basic_rule.xpath_base)
 
     def test_rule_invalid_case(self, rule_constructor, invalid_case):
-        """Check that a rule cannot be instantiated when the case is invalid.
-
-        Todo:
-            Add tests with empty path AND empty xpath_base.
-        """
+        """Check that a rule cannot be instantiated when the case is invalid."""
         xpath_base = 'an xpath'
 
         with pytest.raises(ValueError):
             rule_constructor(xpath_base, invalid_case)
 
     def test_is_valid_for(self, valid_dataset, rule):
-        """Check that a given rule returns the expected result when given a dataset.
-
-        Todo:
-            Maybe too much of a shortcut as can't fully pass until all implementations complete. Possibly the wrong abstraction in the long-term.
-        """
+        """Check that a given rule returns the expected result when given a dataset."""
         assert rule.is_valid_for(valid_dataset)
 
     def test_is_invalid_for(self, invalid_dataset, rule):
-        """Check that a given rule returns the expected result when given a dataset.
-
-        Todo:
-            Maybe too much of a shortcut as can't fully pass until all implementations complete. Possibly the wrong abstraction in the long-term.
-        """
+        """Check that a given rule returns the expected result when given a dataset."""
         assert not rule.is_valid_for(invalid_dataset)
 
+    @pytest.mark.parametrize("junk_data", iati.core.tests.utilities.find_parameter_by_type([], False))
+    def test_is_valid_for_raises_error_on_non_permitted_argument(self, rule, junk_data):
+        """Check that a given rule returns expected error when passed an argument that is not a dataset."""
+        with pytest.raises(AttributeError):
+            rule.is_valid_for(junk_data)
+
+    def test_is_valid_for_raises_error_when_passed_an_etree(self, rule):
+        """Check that an error is raised if an etree is given as an argument instead of a dataset."""
+        with pytest.raises(AttributeError):
+            rule.is_valid_for(iati.core.resources.load_as_tree(iati.core.resources.get_test_data_path('valid_atleastone')))  # use more generic dataset
```

Comment:
Should the comment be a TODO? If so, it should be fixed. If not, it should either be moved and clarified; or removed.